### PR TITLE
AB#634 changed matching algorithm for image.find command (AForge Exha…

### PR DESCRIPTION
**Pull request by wojszub**
…ustiveTemplateMatching algorithm)

### CODE REVIEW

File: G1ANT.Addon.Images/Extensions/ImageExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                var matchings = templateMatching.ProcessImage(source, template);
+                var sourceData = source.LockBits(new Rectangle(0, 0, source.Width, source.Height), ImageLockMode.ReadWrite, source.PixelFormat);
+
+                foreach (var match in matchings)
+                {
+                    Drawing.Rectangle(sourceData, match.Rectangle, Color.White);
+                }
+
+                source.UnlockBits(sourceData);
+                source.Save(@"C:\G1ANTTEST\source.png");
```
Comment (ChrisPrusik) [inline G1ANT.Addon.Images/Extensions/ImageExtensions.cs]: ? :)

Comment (wojszub) [reply G1ANT.Addon.Images/Extensions/ImageExtensions.cs]: removed

File: G1ANT.Addon.Images/Extensions/ImageExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Linq;
+using Image = AForge.Imaging.Image;
+
+namespace G1ANT.Addon.Images
+{
+    public static class ImageExtensions
+    {
+        public static Bitmap OpenFile(this TextStructure filePath)
```
Comment (przegwar) [inline G1ANT.Addon.Images/Extensions/ImageExtensions.cs]: I would suggest changing name of this method to OpenImage as it returns Bitmap

File: G1ANT.Addon.Images/Extensions/ImageExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    return Image.Clone(bitmap, PixelFormat.Format24bppRgb);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Could not open the image file '{filePath.Value}'. Message: {ex.Message}", ex);
+            }
+        }
+
+        public static Bitmap OpenFileOrGetScreenshot(this TextStructure filePath, RectangleStructure screenArea, BooleanStructure relative)
```
Comment (przegwar) [inline G1ANT.Addon.Images/Extensions/ImageExtensions.cs]: I would be more keen on removing "OpenImage" part of this method as it is already implemented as extension. You should decide wether to take screenshot or open image beforehand.

**Pull request merged**